bundles/org.openhab.binding.denonmarantz/src/main/java/org/openhab/binding/denonmarantz/internal/connector/http/DenonMarantzHttpConnector.java

```diff
@@ -336,15 +438,28 @@ private <T> T getDocument(String uri, Class<T> response) throws IOException {
     }
 
     @Nullable
-    private <T, S> T postDocument(String uri, Class<T> response, S request) throws IOException {
+    private <T, S> T postDocument(String uri, Class<T> response, S request)
+            throws TimeoutException, ExecutionException, InterruptedException, HttpCommunicationException {
         try {
             JAXBContext jaxbContext = JAXBContext.newInstance(request.getClass());
             Marshaller jaxbMarshaller = jaxbContext.createMarshaller();
             StringWriter sw = new StringWriter();
             jaxbMarshaller.marshal(request, sw);
 
-            ByteArrayInputStream inputStream = new ByteArrayInputStream(sw.toString().getBytes(StandardCharsets.UTF_8));
-            String result = HttpUtil.executeUrl("POST", uri, inputStream, CONTENT_TYPE_XML, REQUEST_TIMEOUT_MS);
+            Request httpRequest = httpClient.newRequest(uri).timeout(REQUEST_TIMEOUT_MS, TimeUnit.MILLISECONDS)
+                    .content(new StringContentProvider(sw.toString(), StandardCharsets.UTF_8), CONTENT_TYPE_XML)
+                    .method(HttpMethod.POST);
+
+            ContentResponse contentResponse = httpRequest.send();
+
+            String result = contentResponse.getContentAsString();
+            int status = contentResponse.getStatus();
+
+            logger.trace("result of postDocument for uri '{}' (status code {}):\r\n{}", uri, status, result);
+
+            if (!HttpStatus.isSuccess(status)) {
+                throw new HttpCommunicationException("Error retrieving document for uri '{}'", status);
```

Comment:
In the first parameter, I believe you have to provide the uri and not `{}`.
Maybe something else than retrieving ?

Reply:
Nice catch, fixed. I have kept the world "retrieving" for consistency with other debug messages in the class.